extensions/blocks/podcast-player/podcast-player.php

```diff
@@ -177,5 +185,21 @@ function setup_tracks_callback( \SimplePie_Item $episode ) {
 		$track['title'] = esc_html__( '(no title)', 'jetpack' );
 	}
 
+	if ( ! empty( $enclosure->duration ) ) {
+		$track['duration'] = format_track_duration( $enclosure->duration );
+	}
+
 	return $track;
 }
+
+/**
+ * Returns the track duration as a formatted string.
+ *
+ * @param number $duration of the track in seconds.
+ * @return string
+ */
+function format_track_duration( $duration ) {
+	$format = $duration > HOUR_IN_SECONDS ? 'H:i:s' : 'i:s';
+
+	return esc_html( date_i18n( $format, $duration ) );
```

Comment:
I don't know if we need to escape here, since we're escaping late as we display the time.